grid-compose/pkg/types/types.go

```diff
@@ -0,0 +1,70 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type Specs struct {
+	Version  string             `yaml:"version"`
+	Networks map[string]Network `yaml:"networks"`
+	Services map[string]Service `yaml:"services"`
+	Storage  map[string]Storage `yaml:"storage"`
+}
+
+type Network struct {
+	Type string `yaml:"type"`
+}
+
+type Storage struct {
+	Type string `yaml:"type"`
+	Size string `yaml:"size"`
+}
+
+type Service struct {
+	Flist       string       `yaml:"flist"`
+	Entrypoint  string       `yaml:"entrypoint,omitempty"`
+	Environment KVMap        `yaml:"environment"`
```

Comment:
no need for the new type here map[string]string, specially if you don't have any new methods reacting on that type and doesn't provide any type safety